app/providers/DevicesProvider.js

```diff
@@ -46,24 +51,30 @@ function DevicesProvider({ children }) {
         sync: {
           user: realmUser,
           partitionValue: `device=${realmUser.id}`,
-          error: (session, syncError) => {
-            console.error('syncError.name: ', syncError.name);
-            if (syncError.message)
-              console.error('syncError.message: ', message);
-          }
+          newRealmFileBehavior: {
+            type: 'openImmediately'
+          },
+          existingRealmFileBehavior: {
+            type: 'openImmediately'
+          },
+          // WARNING: REMEMBER TO REMOVE THE CONSOLE.LOG FOR PRODUCTION AS FREQUENT CONSOLE.LOGS
+          // GREATLY DECREASES PERFORMANCE AND BLOCKS THE UI THREAD. IF THE USER IS OFFLINE,
+          // SYNCING WILL NOT BE POSSIBLE AND THIS CALLBACK WILL BE CALLED FREQUENTLY.
+          
+          // error: (session, syncError) => {
+          //   console.error(`There was an error syncing the Devices realm. (${syncError.message ? syncError.message : 'No message'})`);
+          // }
```

Comment:
Is there a reason this error handler is commented out?

Reply:
We wanted the developers to actively uncomment this part themselves if they want to use it, so that they are more aware of it and the performance decrease that it may cause.